### PR TITLE
Fix memory leaks

### PR DESCRIPTION
**Pull request by benjohnde**
@gaebel this one fixes the memory leaks as described in #23 and in addition also streamlines the ios tests with our recent changes.

### CODE REVIEW

File: iostests/Sources/Keychain.swift [hunk 1]
```diff
@@ -22,6 +22,6 @@ public struct Keychain {
     }
 
     public static var global: KVault {
-        KVault(serviceName: nil, accessGroup: nil)
```
Comment (gaebel) [inline iostests/Sources/Keychain.swift:25]: If the service name and the access group are both nil, the KVault will query all entries of the app. Therefor this shouldn't get changed.

File: iostests/Sources/Keychain.swift [hunk 2]
```diff
@@ -22,6 +22,6 @@ public struct Keychain {
...
+        KVault()
```
Comment (gaebel) [inline iostests/Sources/Keychain.swift:25]: The scope of this KVault is all elements with the default service name.

Comment (gaebel) [reply iostests/Sources/Keychain.swift:25]: I'm basically checking in the tests if:

- KVault(serviceName: nil, accessGroup: nil) will delete everything
- and KVault() or KVault(serviceName: "blabla", accessGroup: nil) deletes only entries with its service name.

File: iostests/Sources/Keychain.swift (hunk 1 above)
Comment (benjohnde) [reply iostests/Sources/Keychain.swift]: Ah nice, wasn't 100% aware of that functionality. Will revert my changes!

File: iostests/Sources/Keychain.swift (hunk 2 above)
Comment (benjohnde) [reply iostests/Sources/Keychain.swift]: Basically see above!

**Comment by benjohnde** [comment]
@gaebel I just pushed a commit, basically a rollback of my previous test changes!

**Pull request merged**